Dockerfile

```diff
@@ -1,79 +1,119 @@
-    apt-add-repository -y ppa:deadsnakes/ppa && \
-    apt-get update && apt-get upgrade -qy && \
-    apt-get install \
-    language-pack-en \
-    locales \
-    git \
-    libmysqlclient-dev \
-    libssl-dev \
-    build-essential \
-    python3.8-dev \
-    python3.8-distutils \
-    python3.8-venv -qy && \
-    rm -rf /var/lib/apt/lists/*
 
-ENV VIRTUAL_ENV=/edx/app/edx-notes-api/venvs/edx-notes-api
-RUN python3.8 -m venv $VIRTUAL_ENV
-ENV PATH="$VIRTUAL_ENV/bin:$PATH"
+# Make necessary directories and environment variables.
+RUN mkdir -p /edx/var/edx_notes_api/staticfiles
+RUN mkdir -p /edx/var/edx_notes_api/media
+# Log dir
+RUN mkdir /edx/var/log/
 
+RUN virtualenv -p python3.8 --always-copy ${EDX_NOTES_API_VENV_DIR}
+RUN virtualenv -p python3.8 --always-copy ${SUPERVISOR_VENV_DIR}
 
-RUN locale-gen en_US.UTF-8
-ENV LANG en_US.UTF-8
-ENV LANGUAGE en_US:en
-ENV LC_ALL en_US.UTF-8
-ENV EDXNOTES_CONFIG_ROOT /edx/etc
-ENV DJANGO_SETTINGS_MODULE notesserver.settings.yaml_config
+#install supervisor and deps in its virtualenv
+RUN . ${SUPERVISOR_VENV_BIN}/activate && \
+  pip install supervisor==${SUPERVISOR_VERSION} backoff==1.4.3 boto==2.48.0 && \
+  deactivate
+
+COPY requirements/base.txt ${EDX_NOTES_API_CODE_DIR}/requirements/base.txt
+
+RUN pip install -r ${EDX_NOTES_API_CODE_DIR}/requirements/base.txt
+
+# Working directory will be root of repo.
+WORKDIR ${EDX_NOTES_API_CODE_DIR}
+
+# Copy over rest of code.
+# We do this AFTER requirements so that the requirements cache isn't busted
+# every time any bit of code is changed.
+COPY . .
+COPY /configuration_files/edx_notes_api_gunicorn.py ${EDX_NOTES_API_HOME}/edx_notes_api_gunicorn.py
+# COPY /configuration_files/discovery-workers.sh ${DISCOVERY_HOME}/discovery-workers.sh
+# COPY /configuration_files/discovery.yml ${DISCOVERY_CFG}
+COPY /scripts/edx_notes_api.sh ${EDX_NOTES_API_HOME}/edx_notes_api.sh
+# COPY /configuration_files/supervisor.conf ${SUPERVISOR_APP_DIR}/supervisord.conf
+# create supervisor job
+COPY /configuration_files/supervisor.service /etc/systemd/system/supervisor.service
+COPY /configuration_files/supervisor.conf ${SUPERVISOR_CFG_DIR}/supervisor.conf
+COPY /configuration_files/supervisorctl ${SUPERVISOR_VENV_BIN}/supervisorctl
+# Manage.py symlink
+COPY /manage.py /edx/bin/manage.edx_notes_api
+
+# Expose canonical Discovery port
+EXPOSE 18281
+
+FROM app as prod
+
+ENV DJANGO_SETTINGS_MODULE "notesserver.settings.dev"
+
+RUN make static
 
-EXPOSE 8120
-RUN useradd -m --shell /bin/false app
+ENTRYPOINT ["/edx/app/edx_notes_api/edx_notes_api.sh"]
 
-WORKDIR /edx/app/notes
+FROM app as dev
 
-# Copy the requirements explicitly even though we copy everything below
-# this prevents the image cache from busting unless the dependencies have changed.
-COPY requirements/base.txt /edx/app/notes/requirements/base.txt
-COPY requirements/pip.txt /edx/app/notes/requirements/pip.txt
+ENV DJANGO_SETTINGS_MODULE "notesserver.settings.devstack"
```

Comment:
I believe this dockerfile is used in the stage, prod and edge EKS clusters so it's not appropriate to hardcode a devstack file here.